### PR TITLE
EA Forum ToS and CC-BY

### PR DESCRIPTION
**Pull request by oetherington**
NOTE: This PR shouldn't be deployed to prod until 1st December!

This PR adds the requirement in the EA forum that users must accept the new terms of use.

This will be presented to new users who signup:
Screenshot 2022-11-07 at 20 37 08

as well as to existing users when they next create or edit a post:
Screenshot 2022-11-07 at 14 25 31
Screenshot 2022-11-07 at 14 28 47

Failing to accept will result in an error:
Screenshot 2022-11-07 at 14 28 54

Note that users can still create and edit drafts without accepting the TOS, but they won't be able to publish them until they do. They can also edit pre-existing posts without accepting. In both cases, the message will still be shown at the top of the page, but clicking publish will still work.

This PR also includes the /termsOfUse page:
Screenshot 2022-11-07 at 17 54 15

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203325347881828) by [Unito](https://www.unito.io)
┆Link To Task: https://app.asana.com/0/1201302964208280/1203325347881828

### CODE REVIEW

**Comment by oetherington** [comment]
Since @jpaddison3 is out of the PR review rotation at the moment, I'll ping you here in case you have any strong UI opinions (no need to review the code though if you don't have time!)

File: packages/lesswrong/components/posts/PostsAcceptTos.tsx [hunk 1]
```diff
@@ -1,9 +1,18 @@
-import React, { useState } from "react";
+import React, { FC, useState } from "react";
 import { registerComponent, Components } from "../../lib/vulcan-lib";
-import { forumTypeSetting, tosUrlSetting, licenseUrlSetting } from "../../lib/instanceSettings";
+import { forumTypeSetting } from "../../lib/instanceSettings";
 import Button from "@material-ui/core/Button";
 import { gql, useMutation } from "@apollo/client";
 import { useMessages } from "../common/withMessages";
+import { Link } from "react-router-dom";
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/PostsAcceptTos.tsx]: That import restriction test has done so much work.

**Comment by jpaddison3** [comment]
UI: Please pull the button over to the left and make it a checkbox.

File: packages/lesswrong/components/users/NewUserCompleteProfile.tsx [hunk 2]
```diff
@@ -161,6 +171,12 @@ const NewUserCompleteProfile: React.FC<NewUserCompleteProfileProps> = ({ current
       </div>}
       {/* TODO: Something about bio? */}
       <div className={classes.submitButtonSection}>
+        {forumTypeSetting.get() === "EAForum" &&
+          <Typography variant="body1" className={classes.sectionHelperText} gutterBottom>
```
Comment (jpaddison3) [inline packages/lesswrong/components/users/NewUserCompleteProfile.tsx:175]: margin-bottom: 16 seems to make me happy here.

File: packages/lesswrong/components/users/NewUserCompleteProfile.tsx [hunk 3]
```diff
@@ -161,6 +171,12 @@ const NewUserCompleteProfile: React.FC<NewUserCompleteProfileProps> = ({ current
...
+            I agree to the <TosLink />, including my content being available
+            under a <LicenseLink /> license
```
Comment (jpaddison3) [inline packages/lesswrong/components/users/NewUserCompleteProfile.tsx:177]:
```suggestion
            under a <LicenseLink /> license.
```

Match the punctuation on the rest of the page.

**Review by Will-Howard — APPROVED** [review]
LGTM

**Comment by oetherington** [comment]
@jpaddison3 
Screenshot 2022-11-16 at 12 27 16

**Comment by jpaddison3** [comment]
LGTM

**Comment by oetherington** [comment]
Cool - I've added an event to my calendar to deploy on 1st December.

**Pull request merged**